truss/remote/baseten/core.py

```diff
@@ -0,0 +1,60 @@
+import logging
+from typing import IO, Optional, Tuple
+
+import truss
+from truss.remote.baseten.api import BasetenApi
+from truss.remote.baseten.utils.tar import create_tar_with_progress_bar
+from truss.remote.baseten.utils.transfer import multipart_upload_boto3
+from truss.truss_handle import TrussHandle
+
+logger = logging.getLogger(__name__)
+
+
+def exists_model(api: BasetenApi, model_name: str) -> bool:
+    models = api.models()
+    model_id_by_name = {model["name"]: model["id"] for model in models["models"]}
```

Comment:
minor: creating an intermediate map seems overkill for just checking if model_name exists in models. We could just iterate through api.models() and fast return.